tasks/register-runner.yml

```diff
@@ -1,13 +1,28 @@
 ---
-- name: Accept gitlab server self signed cert as valid CA
-  ansible.builtin.shell: >
-    set -o pipefail
+# - name: Accept gitlab server self signed cert as valid CA
+#   ansible.builtin.shell: >
+#     set -o pipefail
+
+#     openssl s_client -connect {{ gitlab_runner_coordinator_url | regex_replace('https?:\/\/([^\/]+)\/?.*', '\\1') }}:443 -showcerts </dev/null 2>/dev/null |
+#     sed -e '/-----BEGIN/,/-----END/!d' | tee {{ tls_ca_file }} >/dev/null
+#   args:
+#     executable: /usr/bin/bash
+#   when: force_accept_gitlab_server_self_signed
 
-    openssl s_client -connect {{ gitlab_server_ip }}:443 -showcerts </dev/null 2>/dev/null |
-    sed -e '/-----BEGIN/,/-----END/!d' | tee {{ tls_ca_file }} >/dev/null
-  args:
-    executable: /usr/bin/bash
+- name: Accept gitlab server self signed cert as valid CA
   when: force_accept_gitlab_server_self_signed
+  block:
+    - community.crypto.get_certificate:
+        host: "{{ gitlab_runner_coordinator_url | regex_replace('https?:\/\/([^\/]+)\/?.*', '\\1') }}"
+        port: 443
+        asn1_base64: true # need to hard code to prevent depreciation warning
+      register: gitlab_server_cert
+    - name: Save the GitLab server self signed cert to the default CA path
+      ansible.builtin.copy:
+        content: "{{ gitlab_server_cert.cert }}"
+        dest: "{{ tls_ca_file }}"
```

Comment:
What if this directory does not exist? The copy mudule is not creating the dir, is it?

Reply:
Since both the source and destination are both files, the copy module will not create the destination file parent folder. However , for the update-ca-trust command to work (I am testing with fedora based OS) the cert needs to be put in /usr/share/pki/ca-trust-source/ or /etc/pki/ca-trust/source/ directories. Otherwise the command will not pick up the cert. https://www.linux.org/docs/man8/update-ca-trust.html So those paths are created when the OS is installed and should be there by default. 

Also for the crypto module to work the user will need the ansible community.crypto.get_certificate collection installed. I was not sure if you wanted to stay away from using any collection or module that is not builtin to ansible by default; so i updated the regex for both ways. Thanks!

Reply:
I'm using Ubuntu, and those paths are not present by default. I guess this role would then fail on my machine without setting a correct `tls_ca_file` variable, what I really want to avoid people have to do.

Reply:
I was thinking about this and we can put the defaults for the linux OS in the OS vars file

Reply:
Never mind. I just realized that this whole block is guarded by `force_accept_gitlab_server_self_signed`. Whoever uses this block anyway have to know what he does.